netty/src/test/java/io/grpc/netty/WriteQueueTest.java

```diff
@@ -61,14 +62,16 @@
 @RunWith(JUnit4.class)
 public class WriteQueueTest {
 
+  private final Object lock = new Object();
+
   @Mock
   public Channel channel;
 
   @Mock
   public ChannelPromise promise;
 
-  private long flushCalledNanos;
   private long writeCalledNanos;
+  private long flushCalledNanos = writeCalledNanos + 1;
```

Comment:
No, `Channel.write()` is thread-safe. We used to use `Channel.write()` but now use `WriteQueue` to improve performance via flush coalescing. (So the "that is flushed as a single unit" part of the JavaDoc of `WriteQueue` is the interesting part.)